src/lib.rs

```diff
@@ -1,319 +1,40 @@
-                }
-            }
-        }
-    }
-
-    false
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use bevy::input::ElementState;
-
-    #[test]
-    fn parse_command() {
-        let expected = ConsoleCommandEntered {
-            command: "my-cmd".to_string(),
-            args: "arg1 arg2".to_string(),
-        };
-
-        let input = "my-cmd arg1 arg2".to_string();
-
-        let result: ConsoleCommandEntered = input.into();
-
-        assert_eq!(result, expected);
-    }
-
-    #[test]
-    fn test_console_key_pressed_scan_code() {
-        let input = KeyboardInput {
-            scan_code: 41,
-            key_code: None,
-            state: ElementState::Pressed,
-        };
-
-        let config = vec![ToggleConsoleKey::ScanCode(41)];
-
-        let result = console_key_pressed(&input, &config);
-        assert!(result);
-    }
-
-    #[test]
-    fn test_console_wrong_key_pressed_scan_code() {
-        let input = KeyboardInput {
-            scan_code: 42,
-            key_code: None,
-            state: ElementState::Pressed,
-        };
-
-        let config = vec![ToggleConsoleKey::ScanCode(41)];
-
-        let result = console_key_pressed(&input, &config);
-        assert!(!result);
-    }
-
-    #[test]
-    fn test_console_key_pressed_key_code() {
-        let input = KeyboardInput {
-            scan_code: 0,
-            key_code: Some(KeyCode::Grave),
-            state: ElementState::Pressed,
-        };
-
-        let config = vec![ToggleConsoleKey::KeyCode(KeyCode::Grave)];
-
-        let result = console_key_pressed(&input, &config);
-        assert!(result);
-    }
-
-    #[test]
-    fn test_console_wrong_key_pressed_key_code() {
-        let input = KeyboardInput {
-            scan_code: 0,
-            key_code: Some(KeyCode::A),
-            state: ElementState::Pressed,
-        };
-
-        let config = vec![ToggleConsoleKey::KeyCode(KeyCode::Grave)];
-
-        let result = console_key_pressed(&input, &config);
-        assert!(!result);
-    }
-
-    #[test]
-    fn test_console_key_right_key_but_not_pressed() {
-        let input = KeyboardInput {
-            scan_code: 0,
-            key_code: Some(KeyCode::Grave),
-            state: ElementState::Released,
-        };
-
-        let config = vec![ToggleConsoleKey::KeyCode(KeyCode::Grave)];
-
-        let result = console_key_pressed(&input, &config);
-        assert!(!result);
+        app.init_resource::<ConsoleConfiguration>()
+            .insert_resource(ConsoleState::default())
+            .add_event::<ConsoleCommandEntered>()
+            .add_event::<PrintConsoleLine>()
+            .add_plugin(EguiPlugin)
+            .add_console_command::<ExitCommand, _, _>(exit_command)
+            .add_console_command::<HelpCommand, _, _>(help_command)
+            // if there's other egui code we need to make sure they don't run at the same time
+            .add_system(console_system.exclusive_system())
```

Comment:
I'm not sure this is needed; don't you take a `ResMut` lock on the `egui` resource?

Reply:
Seems like this isn't needed actually. I left this as-is from before my changes, but it seems like I might've removed the need for it somewhere along the way.

Will remove it in my next commit